Modules/Office365DSC/DSCResources/MSFT_EXOSafeAttachmentRule/MSFT_EXOSafeAttachmentRule.psm1

```diff
@@ -0,0 +1,361 @@
+            {
+                $result += @{
+                    $KeyName = $SafeAttachmentRule.$KeyName
+                }
+            }
+            else
+            {
+                $result += @{
+                    $KeyName = $PSBoundParameters[$KeyName]
+                }
+            }
+
+        }
+        if ('Enabled' -eq $SafeAttachmentRule.State)
+        {
+            # Accounts for Get-SafeAttachmentRule returning 'State' instead of 'Enabled' used by New/Set
+            $result.Enabled = $true
+        }
+        else
+        {
+            $result.Enabled = $false
+        }
+
+        Write-Verbose "Found SafeAttachmentRule $($Identity)"
+        Write-Verbose "Get-TargetResource Result: `n $($result | Out-String)"
+        return $result
+    }
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $Identity,
+
+        [Parameter(Mandatory = $true)]
+        [System.String]
+        $SafeAttachmentPolicy,
+
+        [Parameter()]
+        [System.String]
+        $Comments,
+
+        [Parameter()]
+        [System.Boolean]
+        $Enabled = $true,
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfRecipientDomainIs = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfSentTo = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $ExceptIfSentToMemberOf = @(),
+
+        [Parameter()]
+        [uint32]
+        $Priority,
+
+        [Parameter()]
+        [System.String[]]
+        $RecipientDomainIs = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $SentTo = @(),
+
+        [Parameter()]
+        [System.String[]]
+        $SentToMemberOf = @(),
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+    Write-Verbose 'Entering Set-TargetResource'
+    Write-Verbose 'Retrieving information about SafeAttachmentRule configuration'
+    Write-Verbose "Calling Connect-ExchangeOnline function:"
+    Connect-ExchangeOnline -GlobalAdminAccount $GlobalAdminAccount -CommandsToImport '*SafeAttachmentRule'
+    Write-Verbose "Global ExchangeOnlineSession status:"
+    Write-Verbose "$( Get-PSSession -ErrorAction SilentlyContinue | Where-Object Name -eq 'ExchangeOnline' | Out-String)"
+    try
+    {
+        $SafeAttachmentRules = Get-SafeAttachmentRule
+    }
+    catch
+    {
+        Close-SessionsAndReturnError -ExceptionMessage $_.Exception
+    }
+
+    $SafeAttachmentRule = $SafeAttachmentRules | Where-Object Identity -eq $Identity
+
+    if ( ('Present' -eq $Ensure ) -and (-NOT $SafeAttachmentRule) )
+    {
+        try
+        {
+            NewSafeAttachmentRule -NewSafeAttachmentRuleParams $PSBoundParameters
```

Comment:
I understand that this is a custom function, but please use the default PoSh naming convention verb-noun in the function's name.

Reply:
Requested change completed.